src/containers/MyNdla/Arena/components/FlagContentModalContent.tsx

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2023-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { useTranslation } from 'react-i18next';
+import {
+  ModalBody,
+  ModalCloseButton,
+  ModalHeader,
+  ModalTitle,
+  ModalContent,
+} from '@ndla/modal';
+import { Text } from '@ndla/typography';
+import { useState } from 'react';
+import {
+  FormControl,
+  Label,
+  TextAreaV3,
+  RadioButtonGroup,
+  FieldHelper,
+  FieldErrorMessage,
+} from '@ndla/forms';
+import { ButtonV2 } from '@ndla/button';
+import styled from '@emotion/styled';
+import { fonts, spacing } from '@ndla/core';
+import { useSnack } from '@ndla/ui';
+import { useNewFlagMutation } from '../../arenaMutations';
+import handleError from '../../../../util/handleError';
+
+const MAXIMUM_LENGTH_TEXTFIELD = 64;
+
+const StyledButtonRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-end;
+  gap: ${spacing.small};
+`;
+
+const StyledText = styled(Text)`
+  margin-left: auto;
+`;
+
+const StyledModalBody = styled(ModalBody)`
+  display: flex;
+  flex-flow: column;
+  gap: ${spacing.nsmall};
+`;
+
+const StyledRadioButtonGroup = styled(RadioButtonGroup)`
+  > div {
+    > label {
+      ${fonts.sizes('16px')};
+      font-weight: ${fonts.weight.semibold};
+    }
+  }
+`;
+
+interface FlagContentModalProps {
+  id: number;
+  onClose: () => void;
+}
+
+const FlagContentModalContent = ({ id, onClose }: FlagContentModalProps) => {
+  const [flaggedReason, setFlaggedReason] = useState<string>('spam');
+  const [reportReasonText, setReportReasonText] = useState<string>('');
+  const { addNewFlag } = useNewFlagMutation();
+  const { t } = useTranslation();
+  const { addSnack } = useSnack();
+
+  const sendReport = async () => {
+    try {
+      await addNewFlag({
+        variables: {
+          id,
+          type: 'post',
+          reason: flaggedReason === 'other' ? reportReasonText : flaggedReason,
+        },
+      });
+      addSnack({
+        content:
+          'Takk for at du rapporterer innhold som ikke er i tråd med våre retningslinjer. Rapporten er sendt til vår moderator og vil bli gjennomgått så fort som mulig.',
+        id: 'reportPostAdded',
+      });
+    } catch (err) {
+      const typedError = err as { message?: string };
+      addSnack({
+        content: typedError.message,
```

Comment:
Jeg mistenker jo at dette ikke håndterer i18n

Reply:
Det stemmer! Har ikke fikset oversettelser for feilmeldingene enda, men et forslag var å sjekke for visse ord i feilmeldingen fra nodeBB også velge oversettelse basert på det. Ikke veldig robust, men greit nok inntil videre kanskje 😅

Reply:
Eventuelt bare vise feilmeldingen på det språket som er satt på brukeren i nodeBB som en første løsning, også lage en mer robust løsning på sikt